src/realm/sync/subscriptions.cpp

```diff
@@ -49,130 +49,123 @@ constexpr static std::string_view c_flx_sub_name_field("name");
 constexpr static std::string_view c_flx_sub_object_class_field("object_class");
 constexpr static std::string_view c_flx_sub_query_str_field("query");
 
-inline std::string_view to_string_view(StringData in) noexcept
-{
-    return std::string_view(in.data(), in.size());
-}
-
 } // namespace
 
 Subscription::Subscription(const SubscriptionStore* parent, Obj obj)
-    : m_store(parent)
-    , m_obj(std::move(obj))
+    : m_id(obj.get<ObjectId>(parent->m_sub_keys->id))
+    , m_created_at(obj.get<Timestamp>(parent->m_sub_keys->created_at))
+    , m_updated_at(obj.get<Timestamp>(parent->m_sub_keys->updated_at))
+    , m_name(static_cast<std::string_view>(obj.get<StringData>(parent->m_sub_keys->name)))
+    , m_object_class_name(static_cast<std::string_view>(obj.get<StringData>(parent->m_sub_keys->object_class_name)))
+    , m_query_string(static_cast<std::string_view>(obj.get<StringData>(parent->m_sub_keys->query_str)))
```

Comment:
Ah, you're right. I wasn't sure what the obj.get<StringData> would actually do. It looks like it will call the std::string conversion operator to convert to a std::string which is the new type of these members.